### DISCUSSION TITLE
강사님께 질문 정리

### DISCUSSION THREAD

**Discussion by seokjunKing** [WHATDO-TEAM/whatdo#80] [아이디어 제시]
질문:

박건순

1. 저희는 서버를 galera 클러스터3대, maxscale1대로 고가용성DB를 구축했는데
실무에서는 DB를 어떤식으로 운영하고 또 어떤 방식으로 DB서버가 다운되지않게 대비하는지 궁금합니다.

강신우

1. 데이터 전처리및 적재를 airflwo python operator로 하고자 하는데 이럴 경우 고가용성 구성을 어떻게 해야할지 질문드립니다.

2. 로우데이터를 정규화하여 RDB에 적재하였을때 모델 학습용으로 적절한지, 아니면 다른 용도로 활용하는게 좋을지 질문드립니다

허겸

1. 모델의 재학습 및 배포 과정에서 서비스의 중단 없이 자동화 배포 구현을 위해 CI/CD를 통해 구축을 하려고 합니다 
구현 방법에 대해 조언을 듣고 싶습니다

왕석준

1. 장애 발생 시 Airflow 스케줄러를 안정적으로 복구하기 위해 어떤 설정이나 도구를 사용하는지 궁금합니다.
(현재 airflow를 단일 환경 운영 중인 상황에서 pgrep으로 health check 후 없으면 재실행하는 스크립트를 짜서 쓰는 방법도 괜찮은지 궁금합니다.)

2. 실무에서는 Airflow를 고가용성으로 운영할 때 주로 어떤 방식을 사용하시는지 궁금합니다.
예를 들어 CeleryExecutor, KubernetesExecutor 또는 다른 방식을 많이 쓰시는지 궁금합니다.

**Comment by Kdreamtomaster** [comment]
피드백 및 회의 끝나고 강사님께 추가 피드백 받으면서 RDB (마리아디비) 사용하는 이유를 조금더 살을 붙여봤습니다. 
- 물론 RDB 사용이 대규모 데이터를 다룰때 불리한 점은 있지만 원본 데이터 자체로는 분석이 어려워 데이터 자체 가독성을 높일 필요가 있다
- 스타스키마로 DB 스키마를 짜놓으면 추후 대규모 데이터를 다뤄야 하는 상황이 올때 데이터웨어하우스(스노우플레이크) 로 마이그레이션을 염두해둘 수 있다.
- 관련 자료 https://www.integrate.io/ko/blog/snowflake-schemas-vs-star-schemas-what-are-they-and-how-are-they-different-ko/
